icechunk/src/repository.rs

```diff
@@ -378,8 +380,12 @@ impl Repository {
     /// This will remove the branch reference and the branch history. It will not remove the
     /// chunks or snapshots associated with the branch.
     pub async fn delete_branch(&self, branch: &str) -> RepositoryResult<()> {
-        delete_branch(self.storage.as_ref(), &self.storage_settings, branch).await?;
-        Ok(())
+        if branch != "main" {
```

Comment:
Should we use the default branch constant instead of hardcoding main?

Reply:
done